File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/multiline/comment/direct/Java/Python/s893190600_commented.py
import sys
from collections import defaultdict, deque

class Solver:
    def __init__(self, input_stream, output_stream):
        self.input = input_stream
        self.output = output_stream

    def solve(self):
        H = self.ni()
        W = self.ni()
        C = [['.' for _ in range(W + 2)] for _ in range(H + 2)]
        for i in range(H):
            line = self.ns()
            for j in range(W):
                C[i + 1][j + 1] = line[j]
        if H == 1 and W == 1:
            self.prn("No")
            return
        for i in range(1, H + 1):
            for j in range(1, W + 1):
                if C[i][j] == '#' and C[i - 1][j] != '#' and C[i + 1][j] != '#' and C[i][j - 1] != '#' and C[i][j + 1] != '#':
                    self.prn("No")
                    return
        self.prn("Yes")

    def prn(self, s):
        self.output.write(str(s) + "\n")

    def prr(self, s):
        self.output.write(str(s))

    def ni(self):
        return int(self.input.readline())

    def nl(self):
        return int(self.input.readline())

    def nd(self):
        return float(self.input.readline())

    def ns(self):
        return self.input.readline().strip()

def main():
    input_stream = sys.stdin
    output_stream = sys.stdout
    solver = Solver(input_stream, output_stream)
    solver.solve()

if __name__ == "__main__":
    main()

# End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/multiline/comment/direct/Java/C/s893190600_commented.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAX_SIZE 1024 * 8

typedef struct {
    int key;
    int val;
} PP;

typedef struct {
    int key;
    int val1;
    int val2;
} PPP;

typedef struct {
    long key;
    long val;
} PPL;

typedef struct {
    long key;
    long val[2];
} PPDL;

typedef struct {
    char buffer[MAX_SIZE];
    int pos;
    int readLen;
} MyInput;

MyInput* createMyInput(FILE* stream) {
    MyInput* input = (MyInput*)malloc(sizeof(MyInput));
    input->pos = 0;
    input->readLen = fread(input->buffer, sizeof(char), MAX_SIZE, stream);
    return input;
}

int read(MyInput* input) {
    if (input->pos >= input->readLen) {
        input->pos = 0;
        input->readLen = fread(input->buffer, sizeof(char), MAX_SIZE, stdin);
        if (input->readLen <= 0) {
            fprintf(stderr, "End of file reached.\n");
            exit(EXIT_FAILURE);
        }
    }
    return input->buffer[input->pos++];
}

int nextInt(MyInput* input) {
    int len = 0;
    int c = read(input);
    while (!isdigit(c) && c != '-') {
        c = read(input);
    }
    str[len++] = c;
    while (isdigit(c = read(input))) {
        str[len++] = c;
    }
    str[len] = '\0';
    return atoi(str);
}

long nextLong(MyInput* input) {
    int len = 0;
    int c = read(input);
    while (!isdigit(c) && c != '-') {
        c = read(input);
    }
    str[len++] = c;
    while (isdigit(c = read(input))) {
        str[len++] = c;
    }
    str[len] = '\0';
    return atol(str);
}

double nextDouble(MyInput* input) {
    int len = 0;
    int c = read(input);
    while (!isdigit(c) && c != '-' && c != '.') {
        c = read(input);
    }
    str[len++] = c;
    while (isdigit(c = read(input)) || c == '.') {
        str[len++] = c;
    }
    str[len] = '\0';
    return atof(str);
}

char* nextString(MyInput* input) {
    int len = 0;
    int c = read(input);
    while (isspace(c)) {
        c = read(input);
    }
    str[len++] = c;
    while (!isspace(c = read(input))) {
        str[len++] = c;
    }
    str[len] = '\0';
    return strdup(str);
}

void solve(MyInput* input, FILE* output) {
    int H = nextInt(input);
    int W = nextInt(input);
    char C[H + 2][W + 2];
    char c[W + 2];
    for (int i = 0; i < H; i++) {
        strcpy(c, nextString(input));
        for (int j = 0; j < W; j++) {
            C[i + 1][j + 1] = c[j];
        }
    }
    if (H == 1 && W == 1) {
        fprintf(output, "No\n");
        return;
    }
    for (int i = 1; i <= H; i++) {
        for (int j = 1; j <= W; j++) {
            if (C[i][j] == '#' && C[i - 1][j] != '#' && C[i + 1][j] != '#' &&
                C[i][j - 1] != '#' && C[i][j + 1] != '#') {
                fprintf(output, "No\n");
                return;
            }
        }
    }
    fprintf(output, "Yes\n");
}

int main() {
    MyInput* input = createMyInput(stdin);
    FILE* output = stdout;
    solve(input, output);
    free(input);
    fclose(output);
    return 0;
}
// End of Code
